library/Zend/Filter/Bytes.php

```diff
@@ -0,0 +1,245 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Filter;
+
+use Zend\Filter\Exception\InvalidArgumentException;
+
+final class Bytes extends AbstractFilter
+{
+    const MODE_BINARY = 'binary';
+    const MODE_DECIMAL = 'decimal';
+
+    const TYPE_BITS = 'b';
+    const TYPE_BYTES = 'B';
+
+    /**
+     * A list of all possible filter modes:
+     *
+     * @var array
+     */
+    private static $modes = array(
+        self::MODE_BINARY,
+        self::MODE_DECIMAL,
+    );
+
+    /**
+     * A list of all possible filter types
+     *
+     * @var array
+     */
+    private static $types = array(
+        self::TYPE_BITS,
+        self::TYPE_BYTES,
+    );
+
+    /**
+     * A list of standardized binary prefix formats for decimal and binary mode
+     * @link https://en.wikipedia.org/wiki/Binary_prefix
+     *
+     * @var array
+     */
+    private static $standardizedPrefixes = array(
+        // binary IEC units:
+        self::MODE_BINARY => array('', 'Ki', 'Mi', 'Gi', 'Ti', 'Pi', 'Ei', 'Zi', 'Yi'),
+        // decimal SI units:
+        self::MODE_DECIMAL => array('', 'k', 'M', 'G', 'T', 'P', 'E', 'Z', 'Y'),
+    );
+
+    /**
+     * Default options:
+     *
+     * @var array
+     */
+    protected $options = array(
+        'mode'         => self::MODE_DECIMAL,
+        'type'         => self::TYPE_BYTES,
+        'precision'    => 2,
+        'prefixes'     => array(),
+    );
+
+    /**
+     * @param array $options
+     */
+    public function __construct($options = array())
+    {
+        if (!static::isOptions($options)) {
+            return;
+        }
+
+        $this->setOptions($options);
+    }
+
+    /**
+     * Define the mode of the filter. Possible values can be fount at self::$modes.
+     *
+     * @param string $mode
+     *
+     * @throws InvalidArgumentException
+     */
+    protected function setMode($mode)
+    {
+        $mode = strtolower($mode);
+        if (!in_array($mode, self::$modes)) {
+            throw new InvalidArgumentException(sprintf('Invalid binary mode: %s', $mode));
+        }
+        $this->options['mode'] = $mode;
+    }
+
+    /**
+     * Get current filter mode
+     *
+     * @return string
+     */
+    public function getMode()
```

Comment:
Defining public API that is not covered by an interface is not really a good idea when mixed with `final`, as any coupling with this method will be un-mockable. Consider removing public getters alltogether